### PR TITLE
Add data grid documentation for association filtering

### PR DESCRIPTION
**Pull request by iobuhov**
This PR adds documentation for new (not released yet) data grid feature - association filtering.

Since this new feature change data grid settings, screenshots has to be updated to match new changes (only column settings dialog). If you need project with new data grid, please reach out to web-content team.

### CODE REVIEW

**Comment by ConnorLand** [comment]
Hello @iobuhov , do you have an approximate release date for this feature?

**Comment by iobuhov** [comment]
@ConnorLand, well, I think we will release in a couple of weeks. Probably by the end of this sprint, or at the beginning of next. But, please, don't rush with review, as here more changes is coming. I'll let you know when we will have final docs update.

File: content/en/docs/appstore/modules/data-widgets/data-grid-2.md [hunk 1]
```diff
@@ -170,6 +170,16 @@ This property not available if you enable "Custom content" for the column
 
 {{< figure src="/attachments/appstore/modules/data-widgets/data-grid-2/tooltip-prop.png" alt="Example of wrap text" >}}
 
+### 4.9 Association filter
+
+* **Reference** — this property define association that will be used for filtering. When set, it enable filtering over association with the Drop-down filter. Drop-down filter will prioritize this property over `Attribute` property.
```
Comment (leonardomendix) [inline content/en/docs/appstore/modules/data-widgets/data-grid-2.md]:
```suggestion
* **Reference** — this property defines the association that will be used for filtering. When set, it enables filtering over association with the Drop-down filter. Drop-down filter will prioritize this property over `Attribute` property.
```

File: content/en/docs/appstore/modules/data-widgets/data-grid-2.md [hunk 2]
```diff
@@ -272,10 +282,12 @@ In this section you can select an action to be executed **On change** by the fil
 
 {{< figure src="/attachments/appstore/modules/data-widgets/data-grid-2/dropdown-filter.png" alt="Example of default drop-down filter" >}}
 
-**Drop-down filter** allows users to match enumeration values or a Boolean attribute. To configure the available options when you press the drop-down filter, you can manually add them in the options list or select **Automatic options** to automatically load the values:
+**Drop-down filter** allows users to match enumeration values, Boolean attribute or an association attribute. To configure the available options when you press the drop-down filter, you can manually add them in the options list or select **Automatic options** to automatically load the values:
```
Comment (leonardomendix) [inline content/en/docs/appstore/modules/data-widgets/data-grid-2.md]:
```suggestion
**Drop-down filter** allows users to match enumeration values, Boolean attributes, or an association attribute. To configure the available options when you press the drop-down filter, you can manually add them to the options list or select **Automatic options** to automatically load the values:
```

File: content/en/docs/appstore/modules/data-widgets/data-grid-2.md [hunk 3]
```diff
@@ -272,10 +282,12 @@ In this section you can select an action to be executed **On change** by the fil
...
 
 {{< figure src="/attachments/appstore/modules/data-widgets/data-grid-2/dropdown-filter-settings.png" alt="Example of settings for drop-down filter" >}}
 
+When Drop-down filter used in conjunction with [Association filter](#49-association-filter) data grid configuration, only **Automatic options** supported. Any custom options will be ignored. Please check [Association filter](#49-association-filter) section to learn how to enable association filtering.
```
Comment (leonardomendix) [inline content/en/docs/appstore/modules/data-widgets/data-grid-2.md]:
```suggestion
When Drop-down filter is used in conjunction with [Association filter](#49-association-filter) data grid configuration, only **Automatic options** are supported. Any custom options will be ignored. Please check [Association filter](#49-association-filter) section to learn how to enable association filtering.
```

File: content/en/docs/appstore/modules/data-widgets/data-grid-2.md [hunk 4]
```diff
@@ -412,6 +424,10 @@ The attributes selected here will be used for the matching filter placed inside
 If a filter is being used and its type does not match with any selected attribute, then it will throw an error requesting you to select the correct filter widget.
 {{% /alert %}}
 
+{{% alert color="warn" %}}
+There cannot be more than one filter of the same type in the data grid header. For example, if data grid header already have `Number Filter` adding another `Number Filter` will cause incorrect data filtering.
```
Comment (leonardomendix) [inline content/en/docs/appstore/modules/data-widgets/data-grid-2.md]:
```suggestion
There cannot be more than one filter of the same type in the data grid header. For example, if data grid header already has a `Number Filter` adding another `Number Filter` will cause incorrect data filtering.
```

**Review by leonardomendix — COMMENTED** [review]
Tiny suggestions.

**Comment by iobuhov** [comment]
@ConnorLand, I think my PR is ready to be reviewed 😄

**Review by leonardomendix — APPROVED** [review]

**Comment by ConnorLand** [comment]
> @ConnorLand, I think my PR is ready to be reviewed 😄

Great, will do!

**Comment by iobuhov** [comment]
@ConnorLand, can you please update screenshots in DG section as some property names are changed. Thank you.

**Review by ConnorLand — APPROVED** [review]

**Pull request merged**